### PR TITLE
Sb cross receiver samples

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/samples/README.md [hunk 1]
```diff
@@ -26,3 +26,5 @@ description: Samples for the Azure.Messaging.ServiceBus client library
 - [Managing rules](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample12_ManagingRules.md)
 - [Advanced configuration](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample13_AdvancedConfiguration.md)
 - [Interact with the AMQP message](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample14_AMQPMessage.md)
+- [Mocking Client Types](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample15_MockingClientTypes.md)
+- [Cross-receiver message settlement]
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/samples/README.md:30]: Looks like you missed the link here.

File: sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# Cross Receiver Message Settlement
+
+The message settlement APIs on the `ServiceBusReceiver` require passing in the
+`ServiceBusReceivedMessage`. This can be limiting for scenarios in which the message needs to be persisted and then
+settled across process boundaries. There are two strategies that can be used for settling a message with a different
+receiver. The correct strategy to use will depend on the specific scenario.
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md:6]:
```suggestion
receiver. The recommended strategy depends on the specific application scenario.
```
My intent here was to preserve what you were saying but soften the tone just a bit.

File: sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+## Storing the Entire `ServiceBusReceivedMessage`
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md:8]:
```suggestion
## Storing the entire `ServiceBusReceivedMessage`
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+```C# Snippet:ServiceBusReadReceivedMessage
+AmqpAnnotatedMessage amqpMessage = AmqpAnnotatedMessage.FromBytes(new BinaryData(amqpMessageBytes));
+ServiceBusReceivedMessage rehydratedMessage = ServiceBusReceivedMessage.FromAmqpMessage(amqpMessage, new BinaryData(lockTokenBytes));
+
+var client2 = new ServiceBusClient(connectionString);
+ServiceBusReceiver receiver2 = client2.CreateReceiver(queueName);
+await receiver2.CompleteMessageAsync(rehydratedMessage);
+```
+
+## Storing Only the Lock Token
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md:39]:
```suggestion
## Storing only the lock token
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+If the entire message is not needed when settling the message in a different process, you can simply store off the
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md:41]:
```suggestion
If the entire message is not needed when settling the message in a different process, you can simply preserve the
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+lock token. In the example below, we store off the lock token with the GUID bytes. You can also simply store a
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md:42]:
```suggestion
lock token. In the example below, we store off the lock token using its GUID bytes. You can also simply store a
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+var message = new ServiceBusMessage("some message");
+await sender.SendMessageAsync(message);
+
+ServiceBusReceiver receiver1 = client1.CreateReceiver(queueName);
+ServiceBusReceivedMessage receivedMessage = await receiver1.ReceiveMessageAsync();
+ReadOnlyMemory<byte> lockTokenBytes = Guid.Parse(receivedMessage.LockToken).ToByteArray();
+```
+
+
+In order to rehydrate the message in another process using the lock token, we would do the following:
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md:58]:
```suggestion
In order to rehydrate the message in another process using the lock token, we would do the following:  
```

File: sdk/servicebus/Azure.Messaging.ServiceBus/samples/README.md (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/samples/README.md:30]: Will need to add this after the PR merges.

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/samples/README.md (hunk 1 above)
Comment (jsquire) [reply sdk/servicebus/Azure.Messaging.ServiceBus/samples/README.md:30]: You're probably going to have to strip the brackets.  I think the analyze pass will fail due to being a malformed link.

**Pull request merged**